### PR TITLE
feat(notifications-service): complete revamp

### PR DESCRIPTION
**Pull request by deshmukhmayur**
Complete rewrite of the notifications service

# Explain the feature/fix

- Added email and push notification support with a robust queue
- User subscriptions and subscription management
- Custom content types for notifications
- option to run the queue separately thus increasing scalability

## Does this PR introduce a breaking change

Yes.

The notifications service is completely revamped, with major schema and API changes.

## Screenshots

N/A

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demoed and the design review approved?

### CODE REVIEW

**Review by ghanlohar — APPROVED** [review]
LGTM

File: packages/notifications-service/README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# Notifications Service
+
 # Notifications Microservice
```
Comment (riginoommen) [inline packages/notifications-service/README.md:3]:
```suggestion
## Notifications Microservice
```

**Review by riginoommen — APPROVED** [review]

**Review by riginoommen — CHANGES_REQUESTED** [review]
Just update the readme.md

File: packages/notifications-service/README.md (hunk 1 above)
Comment (deshmukhmayur) [reply packages/notifications-service/README.md:3]: Why `h2`? We're using h1 for all the readme files.

Comment (deshmukhmayur) [reply packages/notifications-service/README.md:3]: Oh wait, that line is duplicated. There should be only 1 heading. I'll remove it.

Comment (deshmukhmayur) [reply packages/notifications-service/README.md]: Updated.

**Review by riginoommen — APPROVED** [review]

**Review by ghanlohar — APPROVED** [review]

**Pull request merged**